pilot/pkg/bootstrap/server.go

```diff
@@ -308,6 +308,7 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 	// is used as the authentication result.
 	authenticators := []security.Authenticator{
 		&authenticate.ClientCertAuthenticator{},
+		&authenticate.XfccAuthenticator{},
```

Comment:
we can't enable this by default or you could just set the header

Reply:
I think we either need
- check identity with one of the other methods, assert it's an identity we configured as "can delegate" and (ie gateway) and then allow the xfcc check
- flag to enable

cc @costinm has been looking into much of the same

Reply:
Right -  XFCC requires double authentication. See my docs on gateway improvements as well.

The request should be authenticated with ClientCert or JWT - and if the gateway doesn't support this and we are running on a 'secure' network, we can use CIDR/IP matching to validate the request is only from a gateway. It is also possible to use NetworkPolicy - but only if all access to Istiod is blocked by policy - except gateway.

In the last case - NetworkPolicy enforcing only possible client to Istiod is the GW - we can have the XfccAuthenticator as in this PR, but we should have some env to indicate 'network policy' mode.

For all other cases - we will need some extra checks, again see my proposal, it is not an istiod-specific problem but all workloads behind a gateway ( including istio gateways behind other LBs) will need this mechanism.

Reply:
The use case here is envoy front ending Istiod and it takes care of TLS termination. So the connection is from localhost. I think if the peer connection is from localhost and Xfcc Authenticator is enabled, we can rely on xfcc authenticator - Do you agree or any concerns?